ocs_ci/ocs/node.py

```diff
@@ -827,3 +827,61 @@ def get_node_az(node):
     """
     labels = node.get().get('metadata', {}).get('labels', {})
     return labels.get('topology.kubernetes.io/zone')
+
+
+def delete_and_create_osd_node_vsphere_upi(
+    osd_node_name, use_existing_node=False
+):
+    """
+    Unschedule, drain and delete osd node, and creating a new osd node.
+    At the end of the function there should be the same number of osd nodes as
+    it was in the beginning, and also ceph health should be OK.
+    This function is for vSphere UPI.
+
+    Args:
+        osd_node_name (str): the name of the osd node
+        use_existing_node (bool): If False, create a new node and label it.
+            If True, use an existing node to replace the deleted node
+            and label it.
+    """
+
+    osd_node = get_node_objs(node_names=[osd_node_name])[0]
+    remove_nodes([osd_node])
+
+    log.info(f"name of deleted node = {osd_node_name}")
+
+    if config.ENV_DATA.get('rhel_workers'):
+        node_type = constants.RHEL_OS
+    else:
+        node_type = constants.RHCOS
+
+    if not use_existing_node:
+        log.info("Preparing to create a new node...")
+        add_new_node_and_label_upi(node_type, 1)
+    else:
+        node_not_in_ocs = get_worker_nodes_not_in_ocs()[0]
+        ocs_labeled_nodes([node_not_in_ocs], node_type)
+
+
+def ocs_labeled_nodes(nodes, node_type):
```

Comment:
please add a docstring.
Also, this function name is not action-oriented. Maybe change to something like `label_node_and_set_selinux_permissions()`

Reply:
Or maybe we should delete `set_selinux_permissions` from the function?

Reply:
you mean take the set_selinux_permissions part out to a separate function? If so, I agree

Reply:
Fixed